libnavigation-util/src/test/java/com/mapbox/navigation/utils/internal/ThreadControllerTest.kt

```diff
@@ -2,14 +2,31 @@ package com.mapbox.navigation.utils.internal
 
 import kotlinx.coroutines.channels.Channel
 import kotlinx.coroutines.channels.ClosedReceiveChannelException
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
 import kotlinx.coroutines.runBlocking
 import org.hamcrest.CoreMatchers.`is`
 import org.junit.Assert.assertFalse
 import org.junit.Assert.assertThat
 import org.junit.Assert.assertTrue
 import org.junit.Test
 
-class ThreadControllerKtTest {
+class ThreadControllerTest {
+
+    @Test
+    fun jobCountValidationNonUIScope() {
+        val maxCoroutines = 10
+        val maxDelay = 100L
+        val jobControl = ThreadController.getIOScopeAndRootJob()
+        (0 until maxCoroutines).forEach {
+            jobControl.scope.launch {
+                delay(maxDelay)
```

Comment:
`= coroutineRule.runBlockingTest`

this will delay the test thread no?

Reply:
oh this change was ThreadControllerKtTest to ThreadControllerTest. up to you to update

Reply:
Actually this doesn't affect the test as the `launch` is async i.e. the `delay` is not blocking.